### ISSUE TITLE
Quick naive fix for sqlalchemy list view spewing transactions

### ISSUE THREAD

**Issue by abramclark** [flask-admin/flask-admin#2238]
Without this, every request of a sqlalchemy list view creates a transaction that's never cleaned up. I don't like how transaction handling is littered throughout, but this is a quick fix that I'd like in master soon.
```
diff --git a/flask_admin/contrib/sqla/view.py b/flask_admin/contrib/sqla/view.py
index 3095e8d3..05addaab 100755
--- a/flask_admin/contrib/sqla/view.py
+++ b/flask_admin/contrib/sqla/view.py
@@ -1082,6 +1082,7 @@ class ModelView(BaseModelView):
                                                                          count_joins,
                                                                          filters)

+        self.session.begin_nested()
         # Calculate number of rows if necessary
         count = count_query.scalar() if count_query else None

@@ -1099,6 +1100,7 @@ class ModelView(BaseModelView):
         if execute:
             query = query.all()

+        self.session.rollback()
         return count, query

     def get_one(self, id):
```

**Comment by caffeinatedMike** [comment]
PRs are welcome. As long as it doesn't lead to any failing tests/regression elsewhere I don't see an issue with including this.

**Comment by abramclark** [comment]
Did I just try to do a PR wrong? I got a permission denied error.

**Comment by caffeinatedMike** [comment]
There's no way for me to tell. Did you fork the repository? I don't see it listed in your [repository list](https://github.com/abramclark?tab=repositories).

**Comment by abramclark** [comment]
We found a better more general fix that doesn't require patching the library:
```
    @app.teardown_appcontext
    def shutdown_session(response_or_exc):
        session.remove()
```
Where session is a `sqlalchemy.orm.scoped_session(...)` that's passed to `flask_admin.Admin(...).add_view(...)`. Ideally this wouldn't be necessary of course.

**Comment by caffeinatedMike** [comment]
Without proof-of-concept code that reproduces what you're talking about we can't verify an issue, let alone fix it.

**Comment by abramclark** [comment]
Indeed. I'm working on a minimal reproduction. It's taking me some time extracting it.

**Comment by jbbqqf** [comment]
Hi! Triaging older issues — looks like this one self-resolved upstream and the original symptom doesn't apply to flask-admin's recommended setup.

Evidence:
- The reporter's [own follow-up](https://github.com/pallets-eco/flask-admin/issues/2238#issuecomment-1088093099) lands on a small, application-level fix: register `@app.teardown_appcontext` to call `session.remove()` on the `scoped_session` passed to `Admin(...).add_view(...)`. No flask-admin patch needed.
- For users on Flask-SQLAlchemy (the recommended integration), this teardown is [already wired up by the extension](https://flask-sqlalchemy.palletsprojects.com/en/3.1.x/contexts/) -- the leak shows up only when a bare `sqlalchemy.orm.scoped_session` is wired into Flask-Admin without the matching teardown.
- The originally-proposed patch (wrapping the count + query in a manual `session.begin_nested()` / `session.rollback()` pair inside `get_list`) was [explicitly noted by the reporter as a quick fix](https://github.com/pallets-eco/flask-admin/issues/2238#issue-1167611075) -- "I don't like how transaction handling is littered throughout, but this is a quick fix" -- and would conflict with users who *want* `get_list` to share their request-scoped transaction state.

If you agree, would you mind closing this one out (with a pointer to the `teardown_appcontext` workaround for anyone landing here from a search)? Happy to be wrong if there's still a current-`master` repro using `Flask-SQLAlchemy`.

---
*Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the references above were verified manually.*

**Comment by ElLorans** [comment]
Only one minor point: I wouldn't say Flask-SQLAlchemy the recommended integration, but it's by far the most used one
